src/Colors/index.js

```diff
@@ -0,0 +1,101 @@
+import { mergeColors, calculateAlpha, isObject, isString } from './utils.js'
+
+let colors = {
+  white: '#ffffff',
+  black: '#000000',
+  red: '#ff0000',
+  green: '#00ff00',
+  blue: '#0000ff',
+  yellow: '#feff00',
+  cyan: '#00feff',
+  magenta: '#ff00ff',
+}
+
+const normalizedColors = {
+  //store for normalized colors
+}
+
+const calculateColor = (value, options) => {
+  let targetColor = normalizeColor(value)
+  if (!isNaN(options)) {
+    options = { alpha: options }
+  }
+  if (options && isObject(options)) {
+    if (!isNaN(options.darken)) {
+      targetColor = mergeColors(targetColor, 0xff000000, 1 - options.darken)
+    }
+
+    if (!isNaN(options.lighten)) {
+      targetColor = mergeColors(targetColor, 0xffffffff, 1 - options.lighten)
+    }
+
+    if (!isNaN(options.opacity)) {
+      options.alpha = options.opacity / 100
+    }
+
+    if (!isNaN(options.alpha)) {
+      targetColor = calculateAlpha(targetColor, options.alpha)
+    }
+  }
+  return targetColor || 0
+}
+
+const normalizeColor = color => {
```

Comment:
I would suggest to rename this function to be a bit more descriptive of what it does. It wasn't instantly clear to me that this function converts hex colors to the argb format

Reply:
changed function name to "normalizeHexToARGB"